src/foam/nanos/crunch/ui/CapableMinMaxCapabilityWizardlet.js

```diff
@@ -104,34 +133,42 @@ foam.CLASS({
       code: function(oldChoices, newChoices) {
         var choiceWizardlets = this.choiceWizardlets;
 
-        var selectedOldChoices = oldChoices.filter(choice => typeof choice[2] === 'object' ? choice[2].get() : choice[2]);
-        var selectedNewChoices = newChoices.filter(choice => typeof choice[2] === 'object' ? choice[2].get() : choice[2]);
+        var selectedOldChoicesNames = oldChoices.map(choice => choice[0]);
+        var selectedNewChoicesNames = newChoices.map(choice => choice[0]);
 
-        var selectedOldChoicesNames = selectedOldChoices.map(choice => choice[0]);
-        var selectedNewChoicesNames = selectedNewChoices.map(choice => choice[0]);
-
-        if ( selectedOldChoices.length > selectedNewChoices.length ){
+        if ( selectedOldChoicesNames.length > selectedNewChoicesNames.length ){
           var deselectedChoiceName = selectedOldChoicesNames.filter(choice => selectedNewChoicesNames.indexOf(choice) == -1)[0]
 
-          var deselectedChoice = selectedOldChoices.filter(choice => choice[0] === deselectedChoiceName)[0];
+          var deselectedChoice = oldChoices.filter(choice => choice[0] === deselectedChoiceName)[0];
 
           var deselectedWizard = choiceWizardlets.filter(wizard => wizard.title === deselectedChoice[0]);
 
           var { targetPayload } =  deselectedWizard[0];
 
-          this.capable.capablePayloads = this.capable.capablePayloads.filter(capablePayload => capablePayload !== targetPayload);
+          this.capable.getCapablePayloadDAO().remove(
+            targetPayload).then(() => {
+              console.log('CANCELLED ' +
+                targetPayload.capability.name
+              );
+            }
+          );
 
-        } else if (selectedOldChoices.length < selectedNewChoices.length ) {
+        } else if (selectedOldChoicesNames.length < selectedNewChoicesNames.length ) {
 
           var selectedChoiceName = selectedNewChoicesNames.filter(choice => selectedOldChoicesNames.indexOf(choice) == -1)[0]
 
-          var selectedChoice = selectedNewChoices.filter(choice => choice[0] === selectedChoiceName)[0];
+          var selectedChoice = newChoices.filter(choice => choice[0] === selectedChoiceName)[0];
           
           var selectedWizard = choiceWizardlets.filter(wizard => wizard.title === selectedChoice[0]);
 
           var { targetPayload } =  selectedWizard[0];
 
-          this.capable.capablePayloads.push(targetPayload);
+          this.capable.getCapablePayloadDAO().put(
+            targetPayload).then(() => {
+              console.log('SAVED ' +
```

Comment:
Do we wanna keep this log here? (the cancelled one I think is useful for now but maybe not saved)